src/main/java/ah/cryptocoin/domain/ApplicationConf.java

```diff
@@ -38,33 +51,81 @@ public static ApplicationConf getApplicationConf() {
 
     private ApplicationConf() {
         conf = this;
-        compareMode = PriceComparison.PREVIOUS;
+        compareMode = PriceComparison.THRESHOLD;
         rateToCompare = new Rate(-1);
-        loadHistoricalRate();
+        loadCBMonitorData();
         Runtime.getRuntime().addShutdownHook(new Thread(){
             @Override
             public void run() {
                 shutdown();
             }
         });
+    }
 
+    private void checkAndCreateAppDataDir() {
+        File f = new File(CbMonitorHome);
+        if (!f.exists()) {f.mkdir();}
+    }
+    private void loadCBMonitorData() {
+        checkAndCreateAppDataDir();
+        loadRateHistory();
+        loadThresholdRate();
+        Agent.register(historicalRate);
     }
 
-    private void loadHistoricalRate() {
-        File f = new File(System.getProperty("user.home")+File.separator+"coinbase-rh.json");
-        if(!f.exists()) {
-            historicalRate = new RateHistory(new ArrayList<Rate>());
-        } else {
-            try {
-                InputStreamReader isr = new InputStreamReader(new FileInputStream(f));
-                Gson gson = new Gson();
-                historicalRate = gson.fromJson(isr, RateHistory.class);
-            } catch (FileNotFoundException e) {
-                e.printStackTrace();
-            }
+    private void loadRateHistory() {
+        try {
+            historicalRate = loadData(RateHistory.class, rateHistoryFile);
+        } catch (Exception e) {
+            historicalRate = new RateHistory();
```

Comment:
Why we are ignoring all the exception.

Reply:
When we cannot historical data i ignore that and give empty history to start recording and save that.